### DISCUSSION TITLE
Can we obtain a photometric result in units of solar luminosity

### DISCUSSION THREAD

**Discussion by 0batsy0** [phoebe-project/phoebe2#780] [Questions]
Before I ran mcmc, I set the following code
![捕获](https://github.com/phoebe-project/phoebe2/assets/130527283/1116f16e-8801-4477-a81a-3e7b9a98f29f)
I hope that when I run MCMC, I don't need to set the luminosity value parameter.After I finish running, I would like to know if I can obtain a luminosity value distribution in solar luminosity units(not 'w'),and how to obtain it.

**Comment by kecnry** [comment]
Unfortunately, `pblum_mode='dataset-scaled'` will not allow you to access the luminosity and should not be used during MCMC since it recomputes the scaling factor automatically within each iteration and does not account for this as a properly-sampled nuisance parameter.  It is meant to be convenient while trying to optimize the other physical parameters without having to worry about the complexities (and computational overhead) of proper flux-scaling.  Doubly unfortunately, luminosities for contact systems are even more difficult to handle than for detached systems.

If you want access to the luminosities, then you have two choices (both of which require light curve observations with absolute fluxes and to properly account for third light, if applicable):

(1) set `pblum_mode='absolute'` and sample over the distance and then [_compute_ the luminosities](http://phoebe-project.org/docs/2.4/api/phoebe.frontend.bundle.Bundle.compute_pblums) from the stellar properties.  This will be a little difficult in conjunction with MCMC to get a resulting distribution though, as you'll need to manually sample the resulting distribution and propagate through this logic.  Although not exactly the same use-case, [this example shows how to get a distribution on the secondary pblum when its is coupled to the primary pblum](http://phoebe-project.org/docs/latest/examples/distribution_pblum).

(2) set `pblum_mode='component-coupled'` and sample over the passband luminosity directly while leaving the distance fixed.  This should then directly give you a posterior distribution on the luminosity of the primary half of the contact binary.  If you're varying over anything that affects the ratio between the two halves, then you'll still need to do something like above to access the total luminosity of the contact system.  If you're not and the ratio remains fixed, then you can probably get away with directly "multiplying" the distribution.


As for units, you can always convert from W to solar luminosity at the very end, or you can [set the default unit of the parameters](http://phoebe-project.org/docs/latest/api/phoebe.parameters.ParameterSet.set_default_unit_all) (see also the [tutorial on parameter units](http://phoebe-project.org/docs/latest/tutorials/units)).  Just keep in mind that the units/normalization of your observation fluxes should be consistent with the units/normalization of your luminosities and distance defined in the model (the parameters don't need to have the same units, but the values should be appropriate given their units).
